src/main/java/com/dtolabs/rundeck/plugin/overthere/OTWinRMFileCopier.java

```diff
@@ -0,0 +1,805 @@
+                message = re.getMessage();
+            } else {
+                message = "WinRM Error: " + re.getMessage();
+            }
+            context.getExecutionListener().log(Constants.ERR_LEVEL, logprompt + "failed: " + message);
+            throw new FileCopierException(message, Reason.WinRMProtocolError);
+        } catch (RuntimeIOException re) {
+            String message = null;
+            if (context.getLoglevel() > 2) {
+                re.printStackTrace(System.err);
+                message = re.getMessage();
+            } else {
+                message = "runtime exception: " + re;
+            }
+            context.getExecutionListener().log(Constants.ERR_LEVEL, logprompt + "failed: " + message);
+            throw new FileCopierException(message, StepFailureReason.IOFailure);
+        }
+
+    }
+
+    @Override
+    public String copyFile(ExecutionContext context, File file, INodeEntry node, String destination) throws FileCopierException {
+
+        String filePath;
+
+        try {
+            InputStream fileStream = new FileInputStream(file);
+            filePath = copyFileStream(context, fileStream, node, destination);
+        } catch (IOException ioe) {
+            throw new FileCopierException(ioe.getMessage(), Reason.FileNotStreamableError);
+        }
+
+        return filePath;
+    }
+
+    @Override
+    public String copyScriptContent(ExecutionContext context, String script, INodeEntry node, String destination) throws FileCopierException {
+
+        String filePath;
+
+        InputStream scriptStream = new ByteArrayInputStream(script.getBytes());
+        filePath = copyFileStream(context, scriptStream, node, destination);
+
+        return filePath;
+    }
+
+    @Override
+    public String copyFileStream(ExecutionContext context, InputStream inputStream, INodeEntry node) throws FileCopierException {
+        return copyFileStream(context, inputStream, node, null);
+    }
+
+    @Override
+    public String copyFile(ExecutionContext context, File file, INodeEntry node) throws FileCopierException {
+
+        String filePath;
+
+        try {
+            InputStream fileStream = new FileInputStream(file);
+            filePath = copyFileStream(context, fileStream, node, null);
+        } catch (IOException ioe) {
+            throw new FileCopierException(ioe.getMessage(), Reason.FileNotStreamableError);
+        }
+
+        return filePath;
+    }
+
+    @Override
+    public String copyScriptContent(ExecutionContext context, String script, INodeEntry node) throws FileCopierException {
+
+        String filePath;
+
+        InputStream scriptStream = new ByteArrayInputStream(script.getBytes());
+        filePath = copyFileStream(context, scriptStream, node, null);
+
+        return filePath;
+    }
+
+    public static enum Reason implements FailureReason {
+        WinRMProtocolError,
+        FileNotStreamableError,
+        IOWriteError,
+    }
+
+
+
+
+    /**
```

Comment:
A large part of the code below (all of it?) is the same as in the NodeExecutor.
Extract it in a dedicated class, and use it. It might be the same as the one I was suggesting at the beginning of the file.